src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -497,12 +498,33 @@ public void run()
         }
 
         /**
-         * Schedules this instance for execution at a fixed rate.
+         * {@inheritDoc}
          */
-        public void schedule()
+        @Override
+        public long getTimeUntilNextRun()
         {
-            this.scheduledFuture = scheduler.scheduleAtFixedRate(
-                this, WAIT_MS , KEY_FRAME_RATE_MS, TimeUnit.MILLISECONDS);
+            synchronized (injectState)
+            {
+                if (!injectState.active
+                    || injectState.numOfKeyframesSent >= MAX_KEY_FRAMES)
+                {
+                    return -1L;
+                }
+
+                // send 3 "waves" of 10 packet bursts every 1 second. This makes
+                // sure that the signaling has propagated and that eventually a
+                // packet gets through SRTP.
+                long delay = 0;
+                if (injectState.numOfKeyframesSent % 10 == 0)
```

Comment:
The idea for waiting an initial WAIT_MS was to wait for actual video packets from the sender, but we risk of clipping the first few seconds of audio, in case the sender is video muted. I think I’m going to remove this initial wait.